compiler/next/include/chpl/uast/Array.h

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright 2021 Hewlett Packard Enterprise Development LP
+ * Other additional copyright holders may be indicated within.
+ *
+ * The entirety of this work is licensed under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef CHPL_UAST_ARRAY_H
+#define CHPL_UAST_ARRAY_H
+
+#include "chpl/queries/Location.h"
+#include "chpl/uast/Expression.h"
+
+namespace chpl {
+namespace uast {
+
+
+/**
+  This class represents an array expression. For example:
```

Comment:
This could be only me as a not-so-much-of-a-compiler-expert, but anything that evaluates to an array can be called an "array expression". I'd much rather if this said "literal", and we made sure that this type corresponds only to the rhs of the example.

(Or I have a big misunderstanding)

Reply:
Reading and thinking more about it, maybe what we call "literal" must be something that can be statically evaluated? Whereas, I can have an array literal that says `[calculateFirstValue(), calculateSecondValue()]` etc.